liquibase-core/src/main/java/liquibase/parser/core/java/JavaChangeLogParser.java

```diff
@@ -0,0 +1,61 @@
+package liquibase.parser.core.java;
+
+import liquibase.Scope;
+import liquibase.change.Change;
+import liquibase.changelog.ChangeLogParameters;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.exception.ChangeLogParseException;
+import liquibase.parser.ChangeLogParser;
+import liquibase.resource.ResourceAccessor;
+
+import java.lang.reflect.Method;
+
+public class JavaChangeLogParser implements ChangeLogParser {
+
+    @Override
+    public DatabaseChangeLog parse(String physicalChangeLogLocation, ChangeLogParameters changeLogParameters, ResourceAccessor resourceAccessor) throws ChangeLogParseException {
+        try {
+            Class changeLogClass = Class.forName(physicalChangeLogLocation.replace("/", "."));
+            Object changeLogInstance = changeLogClass.newInstance();
+
+            DatabaseChangeLog changelog = new DatabaseChangeLog(physicalChangeLogLocation);
+
+            for (Method method : changeLogClass.getMethods()) {
+                ChangeSet changeSetAnnotation = method.getAnnotation(ChangeSet.class);
+                if (changeSetAnnotation != null && !changeSetAnnotation.ignore()) {
+                    String id = changeSetAnnotation.id();
+                    String author = changeSetAnnotation.author();
+                    boolean runAlways = changeSetAnnotation.runAlways();
+
+                    liquibase.changelog.ChangeSet changeSet = new liquibase.changelog.ChangeSet(id, author, runAlways, false, physicalChangeLogLocation, null, null, changelog);
+
+                    if (method.getReturnType().equals(void.class)) {
```

Comment:
Doing some very simple logic around checking the method setup to know what the user is wanting to do with it. "Void" methods wrap that method in a special JavaChange wrapper that will execute the method when needed. I also stubbed out having methods return Change and ChangeSet objects instead. This will be less useful for you right now, but would be useful for liquibase in general and once the RawSqlChange supports couchbase or a new RawN1qlChange class is made it will be good for you too.